M2/include/valgrind/valgrind.h

```diff
@@ -0,0 +1,7165 @@
+/* -*- c -*-
```

Comment:
@jkyang92 How do memcheck.h, valgrind.h work?  Should they be included like they are? What if they change in the future?

Reply:
These are included so that we don't have a direct dependence on valgrind. The interface they interact with is intended to be stable and so in theory we don't have to update this file unless we end up using some newer feature of valgrind.